engine/src/main/java/org/hibernate/search/v6poc/entity/mapping/building/impl/IndexModelBindingContextImpl.java

```diff
@@ -9,29 +9,29 @@
 import java.util.Optional;
 import java.util.Set;
 
-import org.hibernate.search.v6poc.backend.document.model.spi.IndexSchemaElement;
-import org.hibernate.search.v6poc.backend.document.model.spi.IndexSchemaCollector;
+import org.hibernate.search.v6poc.backend.document.model.spi.IndexElementMetadata;
+import org.hibernate.search.v6poc.backend.document.model.spi.IndexCollectorMetadata;
 import org.hibernate.search.v6poc.entity.mapping.building.spi.IndexModelBindingContext;
-import org.hibernate.search.v6poc.entity.model.spi.EngineHandle;
+import org.hibernate.search.v6poc.entity.model.spi.SearchMetadata;
 import org.hibernate.search.v6poc.entity.model.spi.IndexableTypeOrdering;
 import org.hibernate.search.v6poc.entity.model.spi.IndexedTypeIdentifier;
 
 public class IndexModelBindingContextImpl implements IndexModelBindingContext {
 
-	private final IndexSchemaCollector schemaCollector;
+	private final IndexCollectorMetadata schemaCollector;
 	private final IndexModelNestingContextImpl nestingContext;
-	private final EngineHandle engineHandle = new EngineHandle() {
+	private final SearchMetadata searchMetadata = new SearchMetadata() {
```

Comment:
Still find the name a bit blurry but it's definitely better than the old name. Considering it's a blurry notion, the name probably fits perfectly :).